### PR TITLE
Add exif filtering (fixes #2323)

### LINKED ISSUE #2323: wishlist - Option to remove geotags from exif data in images
This can totally be done by hand obviously, but automation.

### PR DESCRIPTION
**Pull request by ralsina**
Missing:

* [x] Any sort of checking it works
* [x] Some images are just copied and bypass EXIF handling!
* [x] Docs

### CODE REVIEW

**Comment by ralsina** [comment]
@Kwpolska if you can take a look and see if this makes sense, much appreciated.

File: nikola/image_processing.py [hunk 1]
```diff
@@ -49,13 +49,52 @@
...
+        """Connect EXIF tag names to numeric values."""
+        if not EXIF_TAG_NAMES:
+            for ifd in piexif.TAGS:
+                for tag, data in piexif.TAGS[ifd].items():
+                    EXIF_TAG_NAMES[tag] = data['name']
+
+    def filter_exif(self, exif, whitelist):
+        """Filter EXIF data as described in the documentation."""
+        self._fill_exif_tag_names()
+        exif = exif.copy()  # Don't modify in-place, it's rude
```
Comment (Kwpolska) [inline nikola/image_processing.py]: Optimize this by putting it *after* the first two ifs.

File: nikola/conf.py.in [hunk 2]
```diff
@@ -576,10 +576,41 @@ GITHUB_COMMIT_SOURCE = True
...
 # If set to True, EXIF data will be copied when an image is thumbnailed or
-# resized.
+# resized. (See also EXIF_WHITELIST)
 # PRESERVE_EXIF_DATA = False
 
+# If you have enabled PRESERVE_EXIF_DATA, this option lets you choose EXIF
+# fields you want to keep in images. (See also PRESERVE_EXIF_DATA)
 #
+# For a full list of field names, please see here:
+# http://www.cipa.jp/std/documents/e/DC-008-2012_E.pdf
```
Comment (Kwpolska) [inline nikola/conf.py.in]: Is there any nicer document available?

**Comment by Kwpolska** [comment]
I’m not an Exif/photo expert, but LGTM as far as the code is concerned.

File: nikola/conf.py.in (hunk 2 above)
Comment (ralsina) [inline nikola/conf.py.in]: I looked, and not really :-(

**Comment by Kwpolska** [comment]
Feel free to merge.

**Comment by ralsina** [comment]
Thanks for checking it out!

**Pull request merged**